### ISSUE TITLE
Quickinfo incorrectly reports "`any`" on `typeof this`

### ISSUE THREAD

**Issue by Zzzen** [microsoft/TypeScript#47063]
# Bug Report

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/LAKApgHgDg9gTgFwAQGMYDsDOyBmMZIC8SA3qEhUugK4C2AXEgAwA05lOAFFAIZw8MkCAJ5QwMHEIAWAS0wA6GrRZIcjEWInS5ASlLtKlNFmQQi2zAG4DhoaLBIAGuY3jJEayFuUA9D+8BSAB6NoauSACaLvZaCLJWob7+gbZBaSFelAC+oDkgQA)

### 💻 Code

```ts
export const foo = {
    num: 0,
    f(param: typeof this.num, f: typeof this) {
        const x = this;
        type X = typeof x;
        //              ^
        type Y = typeof this;
        //              ^^^^
    }
}
```

### 🙁 Actual behavior
move the cursor to `this`, it shows `any`
![image](https://user-images.githubusercontent.com/6630042/145201564-00b8df3b-f5fa-4a48-bc41-7c6c05a1d963.png)

### 🙂 Expected behavior
like `typeof x`, it should show `{ number: number, f... }`
![image](https://user-images.githubusercontent.com/6630042/145201921-456e107c-eac0-4317-ae32-a6908cb3c26e.png)
